src/logic/map_objects/map_object.h

```diff
@@ -282,6 +282,12 @@ class MapObject {
 	 */
 	void set_reserved_by_worker(bool reserve);
 
+	bool is_marked_for_removal(PlayerNumber) const;
```

Comment:
This should be implemented in Immovable - we won't want to remove bobs, buildings, flags, portdocks, ... ever.

Reply:
Done. Also added an overlay for targeted immovables to the drawing code.